File: lib/models/book.dart
class Book {
  final int? id;
  final String title;
  final String author;
  final String isbn;
  final DateTime? publishDate;
  final bool isAvailable;
  final String? description;

  Book({
    this.id,
    required this.title,
    required this.author,
    required this.isbn,
    this.publishDate,
    required this.isAvailable,
    this.description,
  });

  factory Book.fromJson(Map<String, dynamic> json) {
    return Book(
      id: json['id'] as int?,
      title: json['title'] as String,
      author: json['author'] as String,
      isbn: json['isbn'] as String,
      publishDate: json['publishDate'] != null
          ? DateTime.parse(json['publishDate'] as String)
          : null,
      isAvailable: json['isAvailable'] as bool,
      description: json['description'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'author': author,
      'isbn': isbn,
      'publishDate': publishDate?.toIso8601String(),
      'isAvailable': isAvailable,
      'description': description,
    };
  }
}

File: backend/Models/Book.cs
using System.ComponentModel.DataAnnotations;

namespace LibraryAPI.Models;

public class Book
{
    public int Id { get; set; }

    [Required]
    [StringLength(200)]
    public string Title { get; set; } = string.Empty;

    [Required]
    [StringLength(100)]
    public string Author { get; set; } = string.Empty;

    [Required]
    [StringLength(20)]
    public string ISBN { get; set; } = string.Empty;

    public DateTime? PublishDate { get; set; }

    [StringLength(100)]
    public string? Publisher { get; set; }

    [StringLength(50)]
    public string? Category { get; set; }

    public int? PageCount { get; set; }

    [StringLength(1000)]
    public string? Description { get; set; }

    public string? CoverImageUrl { get; set; }

    public int TotalCopies { get; set; } = 1;

    public int AvailableCopies { get; set; } = 1;

    public bool IsAvailable { get; set; } = true;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    // Navigation properties
    public ICollection<BookRental> BookRentals { get; set; } = new List<BookRental>();
}
